spec/prawn/markup/processor/tables_spec.rb

```diff
@@ -77,11 +76,26 @@
               first_row_top - 4 * line].map(&:round))
   end
 
+  describe 'nested headings' do
+    let(:options) do
+      {
+        heading1: { size: 36, style: :bold },
+        heading2: { size: 24, style: :bold_italic }
+      }
+    end
+
+    it 'creates headings inside tables' do
+      processor.parse('<table><tr><td><h1>hello</h1><h2>world</h2></td></tr></table>')
+      expect(text.strings).to eq(['hello', 'world'])
+      expect(left_positions).to eq([first_col_left, first_col_left])
+      expect(top_positions).to eq([first_row_top - 17, first_row_top - 50].map(&:round))
```

Comment:
Why do the headings start at offset 17 and (36 + 14)?

Reply:
Heading top positions are indeed complex, even in the headings specs we use the "magic" values. I'm fine with these hard-coded values here.